packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/FunctionEditorState.ts

```diff
@@ -388,7 +390,11 @@ export class FunctionEditorState extends ElementEditorState {
     this.executionResult = executionResult;
   };
 
-  setFuncRunPromise = (promise: Promise<ExecutionResult> | undefined): void => {
+  setFuncRunPromise = (
+    promise:
```

Comment:
Since we're using this type union so much, would it make sense to define this as it's own type?